submissions/AndriyM/task_js_oop/app.js

```diff
@@ -0,0 +1,131 @@
+const FIELD_CONF = {
+	WIDTH: 400,
+	HEIGHT: 400,
+	CELL_WIDTH: 100,
+	CELL_HEIGHT: 82,
+	WATTER_BLOCK: -5,
+	ZERO_POSITION: 0
+};
+const PLAYER_CONF = {
+	PLAYER_WIDTH: 52,
+	SPRITE: "images/char-boy.png",
+	MOVE_LEFT: "left",
+	MOVE_RIGHT: "right",
+	MOVE_DOWN: "down",
+	MOVE_UP: "up",
+	INITIAL_POSITION: {
+		x: FIELD_CONF.WIDTH / 2,
+		y: FIELD_CONF.HEIGHT
+	}
+};
+
+const ENEMY_CONF = {
+	AVATAR_INDIVIDUAL: [
+		{ ENEMY_POSITION_Y_TOP: 60, ENEMY_SPEED: 150 },
+		{ ENEMY_POSITION_Y_MIDDLE: 145, ENEMY_SPEED: 200 },
+		{ ENEMY_POSITION_Y_BOTTOM: 228, ENEMY_SPEED: 100 }
+	],
+	PADDING_AVATAR_X: 75,
+	PADDING_AVATAR_Y: 50,
+	SPRITE: "images/enemy-bug.png",
+	START_POSITION_X: FIELD_CONF.ZERO_POSITION - FIELD_CONF.CELL_WIDTH,
+	END_POSITION_X: FIELD_CONF.WIDTH + FIELD_CONF.CELL_WIDTH
+};
+const Enemy = function (x, y, speed, player) {
+	this.x = x;
+	this.y = y;
+	this.speed = speed;
+	this.player = player;
+	this.sprite = ENEMY_CONF.SPRITE;
+};
+
+Enemy.prototype.update = function (dt) {
+	this.x += this.speed * dt;
+	if (this.x > ENEMY_CONF.END_POSITION_X) {
+		this.x = ENEMY_CONF.START_POSITION_X;
+	}
+	this.checkCollisions();
+};
+
+Enemy.prototype.checkCollisions = function () {
+	if (
+		this.player.x + ENEMY_CONF.PADDING_AVATAR_X > this.x &&
+		this.player.x < ENEMY_CONF.PADDING_AVATAR_X + this.x &&
+		this.player.y < ENEMY_CONF.PADDING_AVATAR_Y + this.y &&
+		this.player.y + ENEMY_CONF.PADDING_AVATAR_Y > this.y
+	) {
+		alert("Game over! Score: 0");
+		this.player.reset();
```

Comment:
It was good to have a setTimeout. You could keep both setTimeout and meet the requirement.
It was an opportunity to learn through questions.
There is still one. Ask in Students' chat, you will learn one fundamental thing about asynchronous code and use case for async operations in similar situations..